packages/console/src/VncConsole/VncConsole.js

```diff
@@ -1,11 +1,180 @@
+import RFB from '@novnc/novnc/core/rfb';
 
-const VncConsole = () => <div>VncConsole</div>;
+import { noop } from 'patternfly-react';
 
-VncConsole.propTypes = propTypes;
-VncConsole.defaultProps = defaultProps;
+import VncActions from './VncActions';
+
+const CONNECTING = 'connecting';
+const CONNECTED = 'connected';
+const DISCONNECTED = 'disconnected';
+
+/* eslint no-console: ["warn", { allow: ["error"] }] */
+
+class VncConsole extends React.Component {
+  state = { status: CONNECTING };
+
+  componentDidMount() {
+    const {
+      host,
+      port,
+      path,
+      encrypt,
+      resizeSession,
+      viewOnly,
+      shared,
+      credentials,
+      repeaterID,
+      vncLogging,
+      onInitFailed
+    } = this.props;
+
+    NovncLog.init_logging(vncLogging);
+    try {
+      const protocol = encrypt ? 'wss' : 'ws';
+      const url = `${protocol}://${host}:${port}/${path}`;
+
+      const options = {
+        repeaterID,
+        shared,
+        credentials
+      };
+
+      this.rfb = new RFB(this.novncElem, url, options);
+      this.rfb.addEventListener('connect', this.onConnected);
+      this.rfb.addEventListener('disconnect', this.onDisconnected);
+      this.rfb.addEventListener('securityfailure', this.onSecurityFailure);
+      this.rfb.viewOnly = viewOnly;
+      this.rfb.scaleViewport = false; // if the remote session is smaller than HTML container, the view will be centered
+      this.rfb.resizeSession = resizeSession;
+    } catch (e) {
+      onInitFailed && onInitFailed(e);
+      this.rfb = undefined;
+    }
+  }
+
+  onConnected = () => {
+    this.setState({ status: CONNECTED });
+  };
+
+  onCtrlAltDel = e => {
+    if (this.rfb) {
+      this.rfb.sendCtrlAltDel();
+      this.focusVnc(e);
+    }
+  };
+
+  onDisconnected = e => {
+    this.setState({ status: DISCONNECTED });
+    this.props.onDisconnected(e);
+  };
+
+  onSecurityFailure = e => {
+    this.setState({ status: DISCONNECTED });
+    this.props.onSecurityFailure(e);
+  };
+
+  setNovncElem = e => {
+    this.novncElem = e;
+  };
+
+  focusVnc = e => {
+    if (e && e.target && e.target.blur) {
+      e.target.blur();
+    }
+    this.novncElem && this.novncElem.focus();
+  };
+
+  render() {
+    const {
+      textDisconnected,
+      textConnecting,
+      textSendShortcut,
+      textCtrlAltDel
+    } = this.props;
+
+    let status = null;
+    switch (this.state.status) {
+      case CONNECTED:
+        break;
+      case DISCONNECTED:
+        status = (
+          <div className="vnc-console-disconnected">{textDisconnected}</div>
+        );
+        break;
+      case CONNECTING:
+      default:
+        status = <div className="vnc-console-connecting">{textConnecting}</div>;
+    }
+
+    if (!this.novncStaticComponent) {
+      // create just once
+      this.novncStaticComponent = <div ref={this.setNovncElem} />;
+    }
+
+    return (
+      <div className="vnc-console">
```

Comment:
We should allow a className to be added by callers here.

Reply:
ok